### DISCUSSION TITLE
AttributeError: module 'syft' has no attribute 'launch_duet'

### DISCUSSION THREAD

**Discussion by yohannesb** [OpenMined/courses#451] [Course: Foundations of Private Computation]
My name is Yohannes, and I'm writing to you today with an inquiry related to federated learning and the Syft library.

I'm currently exploring the field of federated learning and embarked on my learning journey by following the "Duet Basics" tutorial. After successfully installing Syft, I attempted to launch both the Data Owner and Data Scientist components of Duet, but I encountered the following error:
![sc](https://github.com/OpenMined/courses/assets/7753293/004107c2-61e6-4483-b942-59c33746564b)

I've reviewed the documentation and troubleshooting resources available, but I haven't been able to resolve the issue myself. Would it be possible to seek your assistance in understanding the cause of this error and finding a solution?

I would greatly appreciate any insights or guidance you can offer. Additionally, I'm happy to provide more details about my environment, Syft version 0.8.4, and Microsoft Windows 11 Pro.

**Comment by ShahNewazKhan** [comment]
Hi @yohannesb were you able to find a resolution for this? I am also encountering the same issue, as per the tutorial I tried installing `syft == 0.5.0`:

![image](https://github.com/OpenMined/courses/assets/3241700/fba024aa-5e3f-420c-a4d3-1112c086bf49)

However I ran into an error:

```
(open-mined-courses) ➜  ~ pip install syft==0.5.0
Collecting syft==0.5.0
  Using cached syft-0.5.0-py2.py3-none-any.whl.metadata (20 kB)
INFO: pip is looking at multiple versions of syft to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement torch<=1.8.1,>=1.4.0 (from syft) (from versions: 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.2.0, 2.2.1)
ERROR: No matching distribution found for torch<=1.8.1,>=1.4.0
```

Therefore I install `syft` with out a pinned version, and it installed `syft-0.8.4`, however looks like this version of `syft` does not have the `launch_duet` method:

![image](https://github.com/OpenMined/courses/assets/3241700/804175e6-fbec-4772-b3bf-5aceab2edd5e)

**Reply by ShahNewazKhan** [reply]
Looking briefly into this, AFAICT, `launch_duet` has been deprecated in favor of this [method](https://github.com/OpenMined/PySyft?tab=readme-ov-file#launch-server).

![image](https://github.com/OpenMined/courses/assets/3241700/e2e08634-b4a4-42dd-9572-c14811cdfd77)
